### PR TITLE
chore(core): adds service for private settings

### PR DESCRIPTION
**Pull request by juho-jaakkola**

### CODE REVIEW

File: engine/classes/Elgg/Database/PrivateSettingsTable.php [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+		$entity = $this->entities->get($entity_guid);
+
+		if (!$entity instanceof \ElggEntity) {
+			return false;
+		}
+
+		$name = $this->db->sanitizeString($name);
+
+		return delete_data("DELETE FROM {$this->table}
```
Comment (mrclay) [inline engine/classes/Elgg/Database/PrivateSettingsTable.php]: $this->db

Comment (juho-jaakkola) [inline engine/classes/Elgg/Database/PrivateSettingsTable.php]: fixed

**Comment by mrclay** [comment]
Any reason this is in master instead of 1.x? I think it would cherry-pick clean.

**Comment by juho-jaakkola** [comment]
Good point. I will make a new PR for 1.x.

**Pull request closed without merging**